crates/jit-icache-coherence/Cargo.toml

```diff
@@ -0,0 +1,28 @@
+[package]
+name = "jit-icache-coherence"
```

Comment:
A minor thing, but would it make sense here to call this `wasmtime-jit-icache-coherence` instead and version it along with the other Wasmtime-related crates (so `2.0.0` currently)?

I honestly don't have too many strong opinions here, but absent any other reasons it does seem like a crate that lives in our repo in `crates/` and is published with all the others should probably be named/versioned in line with the others as well.

Reply:
I don't mind the naming, but I'm a bit unsure about publishing.

The main reason is that our `cache_clear` is a lie (on linux), and only works with the `pipeline_flush`. We do mention in the docs that callers must always call `pipeline_flush` but I'm worried that someone might not do that.

Reply:
Ah, so it actually has to be published if it's a dependency of `wasmtime-jit`. If the API safety is a concern, perhaps we could work out how to wrap it in a higher-level interface (e.g. some sort of object with methods to cache-clear and pipeline-flush, and only allows one to pipeline-flush once cache-clear has been invoked at least once)?

Reply:
I've implemented `clear_cache` as unsafe, and added another note on the doc comments for that function. I like the current API as it allows authors to reorder the `clear_cache` and `pipeline_flush` as they see fit. Which should be valid as long as no one tries to execute code in the mean time.

Once we add an actual `clear_cache` implementation it should also be easier to just drop the `unsafe` and keep the same API.

What do you think about that?

Reply:
That sounds good to me!